### PR TITLE
Fixed #430 - Updated share badge detection to treat passive links differently

### LINKED ISSUE #430: No tracker but gate still shown
On the page https://www.bbc.co.uk/news/business-48473259 (and all other news pages), `Facebook Container` shows there being no trackers, but will still put a Gate icon on a button which allows a user to like/ share the story.

It is my understanding that the BBC went to special measures in order not to add *any* FB tracking to our news pages.

We have two FB icons on our page, and only one has the gate icon on it. Is there anything we can do, either by committing to this project, or adjusting code on our page to avoid the gate?

It is important to us as a public organisation that we are clear where data goes and we do not share to 3rd parties within our own platform. I'd like to avoid people believing we have FB tracking.

thoughts?

Screenshot 2019-06-04 at 19 51 29
Screenshot 2019-06-04 at 19 51 19

### PR DESCRIPTION
**Pull request by maxxcrawford**
Fixes #430 

### Mixed Share Test Page
1. Go to the [Mixed share test page](https://maxxcrawford.github.io/social-implementation-examples/facebook/mixed-share/).
1. Confirm that only the anchor links are badged with the text: 
image

### Buzzfeed
1. Go to any list article on [Buzzfeed](https://www.buzzfeed.com/kaylasuazo/things-thatll-help-save-your-day-from-minor-in?origin=hpp).
1. Confirm that the first share link (Big Facebook icon) should be passive. 
image

1. Confirm that the mini share links under each item in the article are blocked.
image

### CODE REVIEW

File: src/content_script.js [hunk 1]
```diff
@@ -218,7 +224,30 @@ function addFacebookBadge (target, badgeClassUId, socialAction) {
       document.querySelector(".fbc-has-badge.js-fbc-prompt-active").classList.remove("js-fbc-prompt-active");
       e.target.parentElement.parentNode.parentNode.classList.remove("active");
     });
-  } else if (socialAction === "share") {
+  } else if (socialAction === "share-passive") {
```
Comment (maxxcrawford) [inline src/content_script.js]: This `else-if` is duplicated below for the share statement. Recommend creating function both can share.

Comment (maxxcrawford) [reply src/content_script.js]: Resolved in 6d0df08431eac294f1cd867114926fc77f994e2e

File: src/content_script.js [hunk 2]
```diff
@@ -48,15 +48,11 @@ const LOGIN_PATTERN_DETECTION_SELECTORS = [
   ".homepage-photo-leader .welcome-buttons div.btn-facebook", // Mixcloud homepage
   ".modal-content .auth-form div.btn-facebook", // Mixcloud login modal
   "[class*='fb-login']" // Default FB class name "fbc-login-button"
-
 ];
 
 // TODO: Disarm click events on detected elements
 const SHARE_PATTERN_DETECTION_SELECTORS = [
-  "[href*='facebook.com/share']",
```
Comment (maxxcrawford) [inline src/content_script.js]: This line was overwriting the L59 entry.

File: src/content_script.js [hunk 3]
```diff
@@ -66,6 +62,12 @@ const SHARE_PATTERN_DETECTION_SELECTORS = [
   "[title='Share on Facebook']" // Medium
 ];
 
+// TODO: Disarm click events on detected elements
+const PASSIVE_SHARE_PATTERN_DETECTION_SELECTORS = [
+  "[href*='facebook.com/dialog/share']", // Share dialog
+  "[href*='facebook.com/sharer']", // Legacy Share dialog
+];
+
```
Comment (maxxcrawford) [inline src/content_script.js]: Both of these URL patterns (when HREFs) work without the Facebook SDK

File: src/content_script.js [hunk 4]
```diff
@@ -250,7 +238,28 @@ function addFacebookBadge (target, badgeClassUId, socialAction) {
 
 }
 
+// Add Event Listener actions/hooks to share badges
+function shareBadgeEventListenerInit(target, htmlBadgeDiv, options) {
+  if (!options.allowClickThrough) {
```
Comment (maxxcrawford) [inline src/content_script.js]: Adding options object to this function to allow passive links to work. Active tracking links will remain in their freezed state.

**Comment by groovecoder** [comment]
Issue in https://github.com/mozilla/contain-facebook/pull/564#pullrequestreview-358395204 is fixed, but the badge icon isn't as reliable on the passive share links now?

**Comment by maxxcrawford** [comment]
@groovecoder It's supposed to only show on hover for passive links. See @e-pang's [comment](https://github.com/mozilla/contain-facebook/issues/430#issuecomment-528074507) about its behavior.

**Pull request merged**